source/geod24/concurrency.d

```diff
@@ -934,76 +1007,616 @@ protected:
         }
     }
 
-    protected class FiberBinarySemaphore
+    /***********************************************************************
+
+        SelectState
```

Comment:
Did you read [the documentation specs](https://dlang.org/spec/ddoc.html) ?
If you want some example of how things would look like, I recommend you look at the agora docs.
In particular, [a simple module like ECC](https://bpfkorea.github.io/agora/agora/common/crypto/ECC.html) would produce [this documentation](https://bpfkorea.github.io/agora/agora/common/crypto/ECC.html).

You can see what looks readable and what doesn't, e.g. if you go to the `Scalar` struct and check the functions, you'll see which ones are well documented, which one is not, and which one should have its description split in two ;)

Reply:
I didn't :) Will take a look now.